File: apps/api/src/main/java/vnpt/vsp/module/package/entity/PackageLicense.java
package vnpt.vsp.module.pkg.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import java.util.UUID;

/**
 * A license entry associated with a course package manifest.
 *
 * Tracks data licensing information as required by Story 4.1 AC-1.
 */
@Entity
@Table(name = "package_license")
public class PackageLicense {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "manifest_id", nullable = false)
    private CoursePackageManifest manifest;

    @NotBlank
    @Column(nullable = false)
    private String name;

    private String url;

    private String spdxId;

    // JPA-required no-arg constructor
    protected PackageLicense() {}

    public PackageLicense(String name, String url, String spdxId) {
        this.name = name;
        this.url = url;
        this.spdxId = spdxId;
    }

    // Getters
    public UUID getId() { return id; }
    public CoursePackageManifest getManifest() { return manifest; }
    public String getName() { return name; }
    public String getUrl() { return url; }
    public String getSpdxId() { return spdxId; }

    // Setters
    public void setManifest(CoursePackageManifest manifest) { this.manifest = manifest; }
}

File: apps/mobile/lib/domain/models/package_license.dart
// Package License Model — VSP Mobile App
//
// License entry for a course package.
// Mirrors PackageLicense from the backend (Story 4.1 PKG-CONTRACT-2).

import 'package:equatable/equatable.dart';

/// A license governing package or content data.
class PackageLicense extends Equatable {
  final String name;
  final String? url;
  final String? spdxId;

  const PackageLicense({required this.name, this.url, this.spdxId});

  factory PackageLicense.fromJson(Map<String, dynamic> json) {
    return PackageLicense(
      name: json['name'] as String,
      url: json['url'] as String?,
      spdxId: json['spdxId'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
    'name': name,
    if (url != null) 'url': url,
    if (spdxId != null) 'spdxId': spdxId,
  };

  @override
  List<Object?> get props => [name, url, spdxId];
}
